src/Common/src/System/Diagnostics/Debug.Unix.cs

```diff
@@ -8,108 +8,118 @@ namespace System.Diagnostics
 {
     // Intentionally excluding visibility so it defaults to internal except for
     // the one public version in System.Diagnostics.Debug which defines
-    // another version of this partial class with the public visibility 
+    // another version of this partial class with the public visibility
     static partial class Debug
     {
-        private static string NewLine => "\n"; 
+#if !MONO
```

Comment:
I was suggesting that you add the following to the Debug.Mono.cs:
```
    static partial class Debug
    {
    static partial class Debug
    {
        static Debug()
        {
          if (PlatformHelper.IsWindows)
          {
            NewLine = Windows.NewLine;
            s_logger = Windows.s_logger;
          }
          else
          {
            NewLine = Unix.NewLine;
            s_logger = Unix.s_logger;
          }
        }
     }
```

That way you can keep the other files as is but you still get the dynamic behavior on Mono.